# Part 2

In [13]:
import pandas as pd

# Load the Excel file to get an overview of its content
excel_file = pd.ExcelFile('/Users/faizantahir/Documents/LUT uni/Term 2/Product Analytics/Module 4/Part 2 Data/524702-XLS-ENG.xlsx')
excel_file.sheet_names
july_data = pd.read_excel(excel_file, sheet_name='July data')
august_data = pd.read_excel(excel_file, sheet_name='August data')

july_data.head(), august_data.head()


(   ID  renew         clv  promotion  tenure     past.60    past.30     past.7  \
 0   1      1  530.597749          0       4   30.853966  24.798221   4.288178   
 1   2      1    0.000000          1      11    4.016695   3.431347   1.715674   
 2   3      0    0.000000          0       6    4.726262   3.766871   1.120972   
 3   4      1  521.434232          0       1  170.520011  74.016872  20.839214   
 4   5      1    0.000000          0       9    0.570165   0.000000   0.000000   
 
    data  international  family  cust_service  os  
 0     0              0       1             1   2  
 1     1              1       0             4   1  
 2     1              1       0             3   1  
 3     1              0       1             3   1  
 4     1              0       0             1   1  ,
       ID  tenure  past.60  past.30  past.7  data  international  family  \
 0  10001       7   144.47    74.44   27.46     1              1       1   
 1  10002       7    45.26    27.46    1.

In [14]:
# Calculate the renewal rates for customers who received promotions vs. those who did not in July data
promotion_renewal_rate = july_data.groupby('promotion')['renew'].mean().reset_index()

promotion_renewal_rate


promotion     renew
0          0  0.745223
1          1  0.815113

## Analysis Results

### Renewal Rates Analysis
- Customers without promotions: Approximately 74.5% renewal rate.
- Customers with promotions: Approximately 81.5% renewal rate.

### Conclusions
- **Problem Ms. Garcia is Facing:** Based on the data, it appears that Ms. Garcia is dealing with issues related to customer retention. The differential in renewal rates suggests that customers receiving promotions are more likely to renew their services.
- **Effectiveness of the July Campaign (A/B Test):** The promotion seems effective, as evidenced by a higher renewal rate among those who received it compared to those who did not.


In [15]:

# Calculate mean values for these factors grouped by renewal status
characteristics_by_renewal = july_data.groupby('renew').mean()

characteristics_by_renewal


ID         clv  promotion     tenure    past.60    past.30  \
renew                                                                        
0      5072.264091    0.000000   0.418182   7.339091  20.443014   9.521965   
1      4980.258846  217.531448   0.520000  11.423974  38.463106  18.721655   

         past.7      data  international    family  cust_service        os  
renew                                                                       
0      2.340868  0.580000       0.400455  0.135000      3.518636  1.676364  
1      4.805058  0.760641       0.396538  0.425769      1.619744  1.678846

## Insights from July Campaign Data Analysis

The table above shows the mean values of various characteristics for customers who did and did not renew their subscriptions. Here are the key insights:

### Customer Lifetime Value (CLV)
- **Renewed**: Average CLV is approximately 217.53.
- **Not Renewed**: CLV is 0, indicating that non-renewing customers may not have been recognized as generating future revenue, or CLV was not calculated because they did not renew.

### Promotion
- A higher proportion of customers who renewed (52%) received promotions compared to those who did not renew (41.8%).

### Tenure
- Customers who renewed have a longer average tenure (about 11.4 months) compared to those who did not (about 7.3 months), suggesting that longer-tenured customers are more likely to renew.

### Usage Patterns (Past 60, 30, 7 days)
- Renewed customers show higher usage in the last 60, 30, and 7 days, indicating that active users are more likely to renew.

### Data Usage
- A higher percentage of renewing customers use data services (76%) compared to non-renewers (58%).

### International Plans and Family Plans
- International plan usage does not vary significantly between groups, but a higher percentage of renewers have family plans (42.6%) compared to non-renewers (13.5%).

### Customer Service Interactions
- Customers who did not renew had more customer service interactions on average (3.5) compared to those who renewed (1.6), suggesting that negative service experiences or unresolved issues may impact renewal decisions.

### Operating System (OS)
- There's no significant difference in OS distribution between renewed and non-renewed customers.

## Recommendations and Strategy for Future Campaigns

### Effectiveness of Retention Campaign
- The retention campaign, as represented by the promotion, seems to have had a positive effect on customer retention rates. The higher renewal rates among promoted customers suggest that such initiatives can effectively enhance customer loyalty.

### Targeting for Future Campaigns
- **Prioritize Customers Based on Tenure and Usage**: Given that longer tenure and higher usage rates correlate with higher renewal rates, Teleko might consider targeting customers who are actively using their services but have not yet reached a long tenure.
- **Focus on Data Users and Family Plans**: Since these features are associated with higher renewal rates, promoting special offers or enhancements on data packages and family plans could be effective.
- **Address High Interaction Customers**: Implementing better support for customers with high numbers of service calls might prevent churn.

### Selective Retention Incentives
- Not all customers should receive the same retention incentives. Tailored promotions based on customer usage patterns, tenure, and service needs can be more cost-effective and impactful.

### Estimating Impact of Targeting Strategy
- To estimate the retention rate and CLV impact of this targeting strategy, a more detailed predictive modeling approach using machine learning could be applied. This would allow Teleko to simulate different scenarios and predict outcomes based on historical data trends.
